assets/hip-1056/protobuf/stream/record_file_item.proto

```diff
@@ -0,0 +1,92 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.block.stream.protoc";
+// <<<pbj.java_package = "com.hedera.hapi.block.stream">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "basic_types.proto";
+import "timestamp.proto";
+
+/**
+ * A Block Item for record files.
+ *
+ * A `RecordFileItem` contains data produced before the innovation of the
+ * Block Stream, when data was stored in files and validated by individual
+ * signature files rather than a block proof.<br/>
+ * This item enables a single format, the Block Stream, to carry both
+ * historical and current data; eliminating the need to search two sources for
+ * block and block chain data.<br/>
+ * Any block containing this item requires special handling.
+ * - The block SHALL have a `BlockHeader`.
+ *    - Some fields in the `BlockHeader` may be interpreted differently, and
+ *      may depend on when the original record file was created.
+ * - The block SHALL NOT have a `BlockProof`.
+ * - The block SHALL end with an `AddressBookProof`, which is only used for
+ *   `RecordFileItem` blocks.
+ * - The block SHALL contain _exactly one_ `RecordFileItem`.
+ * - The block SHALL NOT contain any content item other than a `RecordFileItem`.
+ * - The content of the `RecordFileItem` MUST be validated using the
+ *   signature data and content provided herein according to the
+ *   process used for Record Files prior to the creation of Block Stream.
+ *    - This block item only replaces the requirement to read several
+ *      individual files from cloud storage services.
+ * - The address book relevant to a particular record file SHALL be available
+ *   separately as an `AddressBookProof` item.
+ */
+message RecordFileItem {
+    /**
+     * The consensus time the record file was produced for.<br/>
+     * This comes from the record file name.
+     */
+    proto.Timestamp creation_time = 1;
+
+    /**
+     * The contents of a record file.<br/>
+     * The first 4 bytes are a 32bit int little endian version number.
+     * The versions that existed are 2,3,5 and 6.
+     */
+    bytes record_file_contents = 2;
+
+    /**
+     * The contents of sidecar files for this block.<br/>
+     * Each block can have zero or more sidecar files.
+     */
+    repeated SidecarFile sidecar_file = 3;
+
+    /**
+     * A collection of RSA signatures from consensus nodes.<br/>
+     * These signatures validate the hash of the record_file_contents field.
+     */
+    repeated bytes record_file_hash_signatures = 4;
```

Comment:
shouldn't we also include information that which singature file (as bytes) comes from which node?

Reply:
Good point, we might need to define the format of these a bit better.
@jasperpotts is working on the tool to build these; any thoughts on this, Jasper?